### PR TITLE
CodeGenerator: Assign stack parameters to virtual registers

### PR DESCRIPTION
**Pull request by Kha**
Restricts those pesky ActivationRecordOperands to a single location in
CodeGenerator

### CODE REVIEW

File: src/main/scala/mjis/CodeGenerator.scala [hunk 1]
```diff
@@ -102,6 +103,10 @@ class CodeGenerator(a: Unit) extends Phase[AsmProgram] {
           n => function.prologue.instructions += Mov(RegisterOperand(reg, n.getMode.getSizeBytes), regOp(n))
         }
 
+      // load used stack parameters into virtual registers early and in a fixed order
+      for ((idx, n) <- usedStackParams.toSeq.sortBy(_._1))
+        function.prologue.instructions += Mov(ActivationRecordOperand(8 * (idx + 1 /* return address */), 8), regOp(n))
```
Comment (jspam) [inline src/main/scala/mjis/CodeGenerator.scala]: I guess the sizeBytes of the AR operand should be `regOp(n).sizeBytes`.

File: src/main/scala/mjis/CodeGenerator.scala [hunk 2]
```diff
@@ -321,8 +326,9 @@ class CodeGenerator(a: Unit) extends Phase[AsmProgram] {
     private def getOperand(node: Node): Operand = getCanonicalNode(node) match {
       case n : nodes.Const => ConstOperand(n.getTarval.asInt(), n.getMode.getSizeBytes)
       case n @ ProjExtr(ProjExtr(_, nodes.Start.pnTArgs), argNum) =>
-        if (argNum < ParamRegisters.length) regOp(n) else ActivationRecordOperand(
-        8 * (argNum - ParamRegisters.length + 1 /* return address */), n.getMode.getSizeBytes)
+        if (argNum >= ParamRegisters.length)
```
Comment (jspam) [inline src/main/scala/mjis/CodeGenerator.scala]: The code to fill `usedParamRegisters` should be moved here too.

Comment (Kha) [inline src/main/scala/mjis/CodeGenerator.scala]: We can actually merge the two maps, even better

**Comment by jspam** [comment]
:white_check_mark:  9 additions and 14 deletions. Your contribution to `cloc` minimization is greatly appreciated.

Also, LGTM.

**Pull request merged**